core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java

```diff
@@ -98,8 +110,30 @@ private String getWhereClause(List<ScriptFilter> scriptFilters, boolean onlyLate
         return " WHERE " + filterStatements + (!filterStatements.isEmpty() && onlyLatestVersions ? " and " : "") + onlyLatestVersionStatement;
     }
 
-    private String getLimitAndOffsetClause(Pageable pageable) {
-        return pageable.isUnpaged() ? " " : " limit " + pageable.getPageSize() + " offset " + pageable.getOffset() + " ";
+    /**
+     * This method provide an Order by statement to order the ScriptAndScriptVersionTable
+     *
+     * @param pageable - pageable object containing or not the order object
+     * @return a String containing the ORDER BY statement
+     */
+    private String getOrderByClause(Pageable pageable) {
+        if (pageable.getSort().isUnsorted()) return " ";
+        List<String> sorting = pageable.getSort().stream().map(order -> {
+            // add further sort on the ScriptAndScriptVersionTable here
+            if (order.getProperty().equalsIgnoreCase("NAME")) {
+                return "script_designs.SCRIPT_NM" + " " + order.getDirection();
+            } else if (order.getProperty().equalsIgnoreCase("VERSION")) {
+                return "versions.SCRIPT_VRS_NB" + " " + order.getDirection();
+            } else {
+                return null;
+            }
+        })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        if (sorting.isEmpty()) {
```

Comment:
It would be better to always have a default sorting for the Front-End, e.g by ID and ASC

Reply:
when no sorting is provided, pagination is indeed not deterministic. We can have a discussion on whether or not we should make it order by default on ID or let it up to the user to decide. I created a dedicated issue for this #201